src/Symfony/Component/AssetMapper/ImportMap/ImportMapUpdateChecker.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Symfony\Component\AssetMapper\ImportMap\Resolver\PackageResolverInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ImportMapUpdateChecker
+{
+    private const URL_PACKAGE_METADATA = 'https://registry.npmjs.org/%s';
+
+    public function __construct(
+        private readonly ImportMapConfigReader $importMapConfigReader,
+        private readonly PackageResolverInterface $resolver,
+        private readonly HttpClientInterface $httpClient,
+    ) {
+    }
+
+    /**
+     * @param string[] $packages
+     *
+     * @return PackageUpdateInfo[]
+     */
+    public function getAvailableUpdates(array $packages = []): array
+    {
+        $entries = $this->importMapConfigReader->getEntries();
+        $updateInfos = [];
+        foreach ($entries as $entry) {
+            if (null === $entry->url) {
+                continue;
+            }
+            if (ImportMapType::CSS === $entry->type) {
+                continue;
+            }
+            if (\count($packages) && !\in_array($entry->importName, $packages, true)) {
```

Comment:
I think there's a problem here - it's a bit annoying, tbh, and I deal with it in other spots as well. `$entry->importName` may be a bare package name or it may be a package + path - e.g. `chart.js/auto` (`chart.js` is the package name and `auto` is a path within the package).

So, we need to do some work to split the package from the path. I do it in my PR - https://github.com/symfony/symfony/pull/51786/files#diff-1a5c8dfbbfec4f3e0bc1fd209545bb56a483fa92e2212f0a3a99c8137468eb35R224 - but perhaps we should put this inside of `ImportMapEntry`? A new `getPackageName()` method and a `getPath()` method?

Reply:
Yes I think having this inside the `ImportMapEntry`  would make sense, thought that would mean being able to parse these infos in `ImportMapConfigReader::getEntries`. No problem adding an `PackageResolverInterface` dependency to `ImportMapConfigReader` ?

Reply:
Good news - I don't think this parsing is dependent on the resolver. The string we're parsing is something like `chart.js/auto`, which is a standard string - it's even what you would import, if you were in Node - e.g. `import 'chart.js/auto'`. 

So, I like your idea of putting this parsing logic into `ImportMapConfigReader::getEntries()`.

Reply:
Yes you're right, the package name format should always stay consistent :smiley:
So just to make sure we are clear : when creating an `ImportMapEntry` we keep the `importName` unchanged (ie: name of the index in `importmap.php`) and we add 2 new properties `packageName` and `filePath` to the `ImportMapEntry` class ?

```
// ImportMapConfigReader.php
public function getEntries(): ImportMapEntries
{
    // ...
    [$packageName, $filePath] =  self::splitPackageNameAndFilePath($importName);

    $entries->add(new ImportMapEntry(
        $importName,
        // ...
        packageName: $packageName,
        filePath: $filePath,
    ));
    // ...
}
```

And that raises another question :
We may have the case where we have a package in different versions depending on the files in the importMap.
For instance
```
...
'bootstrap' => [
	'url' => 'https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/+esm',
],
'bootstrap/dist/css/bootstrap.min.css' => [
	'url' => 'https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css',
	'type' => 'css',
],
...
```
I can't really see why a user would need to do this, but it makes me wonder what should be the grouping rule : the packageName (only one line displayed by the command for 'bootstrap') or the importName (two lines, 'bootstrap' and 'bootstrap/dist/css/bootstrap.min.css')

Reply:
Yes, this is what I’m thinking 👍. Though perhaps we should do this in a separate PR to keep things moving - that change will impact various parts of the code. 

For your last question, in the command, we should render the key used in the importmap (so the importName) so that users know exactly which entry is the problem. Then they could have multiple versions like you mentioned and we could accurately report that one has a problem but another does not.

Reply:
Alright I've rebased and made the changes as suggested. I can open a separate PR for the 'packageName' and 'FilePath' if you want.

Reply:
You have already (nicely) added the new 2 properties to `ImportMapEntry` 🥳 . So the only remaining things to do to accomplish this are to find all usages of `->importName` that need to be updated. Not all need to be updated - sometimes we DO want to reference the "import name" and other times we want to reference the "package name". I did a `git grep '\->importName'` to find them.

I thought about having you include them in THIS PR, but at least on instance (in `ImportMapAudit`) looks complex. So let's finish this PR with the 2 new properties, get it merged, then go find and update all of the `$entry->importName` items that need it.